src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -315,6 +325,13 @@ protected virtual void Initialize()
                     TraceWriter = new ConsoleTraceWriter(hostTraceLevel);
                 }
 
+                ConfigureLoggerFactory(settingsManager, metricsLogger);
+
+                // Use the startupLogger in this class as it is concerned with startup. The public Logger is used
+                // for all other logging after startup.
+                _startupLogger = ScriptConfig.HostConfig.LoggerFactory.CreateLogger(LogCategories.Startup);
```

Comment:
Why do we have to switch on logger instance for different categories? Why isn't a category/scope for the logger used to do this instead?